### PR TITLE
fix: add partition alloc check for MacOS 13/14+

### PR DESCRIPTION
**Pull request by VerteDinde**
#### Description of Change

This PR modifies the existing partition alloc ref count size check to account for any version of MacOS higher than 13.0, rather than using the hardcoded 13/14 values. This should allow for easier testing as beta releases come out in the future.

This patch is currently being added upstream and can be removed when we take in the upstream change.

#### Checklist

- [x] PR description included and stakeholders cc'd
- [x] `npm test` passes
- [x] tests are [changed or added](https://github.com/electron/electron/blob/main/docs/development/testing.md)
- [x] relevant documentation, tutorials, templates and examples are changed or added
- [x] [PR release notes](https://github.com/electron/clerk/blob/main/README.md) describe the change in a way relevant to app developers, and are [capitalized, punctuated, and past tense](https://github.com/electron/clerk/blob/main/README.md#examples).

#### Release Notes

Notes: Fixed a partition alloc ref count check for higher MacOS versions.

### CODE REVIEW

**Review by nornagon — APPROVED** [review]

**Comment by deepak1556** [comment]
Partitionalloc for macOS is only enabled since >= 28-x-y, removed other backport tags.

**Comment by deepak1556** [comment]
I don't have access to the new bug so not sure of the details but based on https://crbug.com/1378822 and assuming it to be same issue on newer versions. We also had similar report from macOS 11 and 12 in https://electronhq.slack.com/archives/CB6CG54DB/p1701968889249799, are they different issues ?

File: patches/chromium/enable_partition_alloc_ref_count_size.patch [hunk 1]
```diff
@@ -0,0 +1,34 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: VerteDinde <vertedinde@electronjs.org>
+Date: Wed, 13 Dec 2023 12:53:11 -0800
+Subject: fix: enable partition alloc ref count size for MacOS 13+
+
+Modifies an existing check to check for all versions of MacOS 13
+or higher, including MacOS 14 and potential beta versions.
+
+This patch is currently being added upstream, and can be removed
+when we integrate the upstream changes into Electron.
```
Comment (ckerr) [inline patches/chromium/enable_partition_alloc_ref_count_size.patch:10]: Should include a link, if available

**Review by ckerr — APPROVED** [review]

File: patches/chromium/enable_partition_alloc_ref_count_size.patch (hunk 1 above)
Comment (VerteDinde) [reply patches/chromium/enable_partition_alloc_ref_count_size.patch:10]: Agreed - I went looking and wasn't able to find an existing CL, but I know a version is incoming. I'll keep an eye out and link for it when I see it 👍

**Comment by VerteDinde** [comment]
@deepak1556 That's a good question 🤔 I don't want to lump in 11 and 12 now and potentially introduce a new issue, but let me look into that reported crash and see if there's an upstream issue as a follow up. It does seem like it could be the same bug based on the stacktrace

**Comment by deepak1556** [comment]
thanks!

**Review by codebytere — APPROVED** [review]

**Pull request merged**